kroxylicious-runtime/src/test/java/io/kroxylicious/proxy/internal/codec/ResponseDecoderTest.java

```diff
@@ -179,7 +183,37 @@ void supportsFallbackToApiResponseV0() {
         // then
         assertThat(objects)
                 .singleElement()
-                .extracting("body")
-                .isEqualTo(new ApiVersionsResponseData().setErrorCode(Errors.UNSUPPORTED_VERSION.code()));
+                .isInstanceOfSatisfying(DecodedResponseFrame.class, frame -> {
+                    ApiVersionsResponseData.ApiVersionCollection expectedApiKeys = new ApiVersionsResponseData.ApiVersionCollection();
+                    expectedApiKeys.add(new ApiVersionsResponseData.ApiVersion().setApiKey(ApiKeys.API_VERSIONS.id).setMinVersion((short) 0).setMaxVersion((short) 3));
+                    assertThat(frame.body()).isEqualTo(new ApiVersionsResponseData().setErrorCode(Errors.UNSUPPORTED_VERSION.code()).setApiKeys(expectedApiKeys));
+                    // the apiVersion on the decoded frame is used by the proxy at encoding time when we forward it to the downstream
+                    // we need to ensure that we forward on v0 bytes as some clients will infer from the 35 error code that a v0 body
```

Comment:
more accurately this should say the proxy should behave as per KIP-511, the clients are behaving correctly